docs/proposals/resource-quota-scoping.md

```diff
@@ -0,0 +1,264 @@
+and internally, they set the request as a factor of the limit to control for 
+over-commit or based on known-historical trends.  In this environment, the
+`cluster-admin` is interested in exposing a quota to their end-users that maps
+to their desired limit instead of their request.
+
+### Ability to quota to control footprint
+
+The current `ResourceQuota` document does not allow the ability
+to quota best-effort pods separately from pods with resource
+guarantees.  The `cluster-admin` wants to control footprint
+to fairly share the nodes, but not restrict the ability to
+run best-effort pods.  By definition, a best-effort pod has
+zero footprint.  As a result, the `cluster-admin` requires
+the ability to control the max number of active best-effort pods.
+
+### Ability to quota long-running vs bounded-duration compute resources
+
+The `cluster-admin` may want to quota end-users separately
+based on long-running vs bounded-duration compute resources.
+
+For example, a `cluster-admin` may offer more compute resources
+for long running pods that are expected to have a more permanent footprint
+on the node than bounded-duration pods.
+
+A classic example is a PaaS deployment where the `cluster-admin` may 
+allow a separate budget for pods that run their web application vs pods that
+build web applications.
+
+Another example is providing more quota to a database pod than a
+pod that performs a database migration.
+
+## Use Cases
+
+* As a cluster administrator, I want the ability to quota
+ * compute resource requests
+ * compute resource limits
+ * compute resources for terminating vs non-terminating workloads
+ * compute resources for best-effort vs non-best-effort pods
+
+## Proposed Change
+
+### New quota tracked resources
+
+Support the following resources that can be tracked by quota.
+
+| Resource Name | Description |
+| ------------- | ----------- |
+| cpu | total cpu requests (backwards compatibility) |
+| cpu.request | total cpu requests |
+| cpu.limit | total cpu limits |
+| memory | total memory requests (backwards compatibility) |
+| memory.request | total memory requests |
+| memory.limit | total memory requests |
+
+### Resource Quota Scopes
+
+Add the ability to associate a set of `scopes` to a quota.
+
+A quota will only measure usage for a `resource` if it matches
+the intersection of enumerated `scopes`.
+
+Adding a `scope` to a quota limits the number of resources
+it supports to those that pertain to the `scope`.  Specifying
+a resource on the quota document outside of the allowed set 
+would result in a validation error.
+
+| Scope | Description |
+| ----- | ----------- |
+| Terminating | Match `kind=Pod` where `spec.restartPolicy in (Never, OnFailure)` |
+| NonTerminating | Match `kind=Pod` where `spec.restartPolicy in (Always)` |
+| BestEffort | Match `kind=Pod` where `status.qualityOfService in (BestEffort)` |
+| NonBestEffort | Match `kind=Pod` where `status.qualityOfService not in (BestEffort)` |
+
```

Comment:
I notice I have a typo, but I will add an example to make this clear.  I had examples in an earlier draft that got removed when I refactored.  A quota with scope=BestEffort would not allow it to track CPU or Memory.  It would only match pods that have best effort quality of service which led proposed cgroup hierarchy would offer no guarantees.